viewing_party/party.py

```diff
@@ -1,23 +1,156 @@
+        except KeyError:
+            genre_counts[movie_genre] = 1
+    max_watch_count = 0
+    for genre, watch_count in genre_counts.items():
+        if watch_count > max_watch_count:
+            max_watch_count = watch_count
+            max_genre = genre
+    return max_genre
+'''
+This function takes in 1 value, a dictionary called user_data. We compare the movies under the key "watched" with the movies under
+each friend's "watched" lists. We retun a list of movies that are unique to the user. 
+'''
+def get_unique_watched(user_data):
+    unique_titles = []
+    for movie in user_data["watched"]:
+        friend_watched = False
+        for friend in user_data["friends"]:
+            for friends_movie in friend["watched"]:
+                if friends_movie["title"] == movie["title"]:
+                    friend_watched = True
+                    break
+        if not friend_watched:    
+            unique_titles.append(movie)
+    return unique_titles   
+'''
+This function takes in 1 value, a dictionary called user_data. We compare the movies under the key "watched" with the movies under
+each friend's "watched" lists. We retun a list of movies that are unique to the friends. 
+'''
+def get_friends_unique_watched(user_data):
+    unique_titles = []
+    unique_title_names = []
+    for friend in user_data["friends"]:
+        for friends_movie in friend["watched"]:
+            user_watched = False
+            for movie in user_data["watched"]:
+                if friends_movie["title"] == movie["title"]:
+                    user_watched = True
+                    break
+            if not user_watched and friends_movie["title"] not in unique_title_names:
+                unique_titles.append(friends_movie)
+                unique_title_names.append(friends_movie["title"])
+    return unique_titles     
+'''
+This function takes in 1 value, a dictionary called user_data. We use the function get_friends_unique_watched to identify
+the movies unique to the friends (unwatched by the user), and then identify which of those movies are on hosts that the user is subscribed to. We return the
+ movies that the user hasn't seen and also has access to. 
+'''
+def get_available_recs(user_data):
+    rec_movies = []
+    user_unwatched = get_friends_unique_watched(user_data)
+    for movie in user_unwatched:
+        if movie["host"] in user_data["subscriptions"]:
+            rec_movies.append(movie)
+    return rec_movies
+'''
+This function takes in 1 value, a dictionary called user_data. We use the function get_most_watched_genre to identify which genre the
+user watches the most, and use the get_friends_unique_watched to identify the movies that the user has not seen. Out of the movies that are
+unique to the friends, we return a list of the ones that match the most watched genre. 
+'''
+def get_new_rec_by_genre(user_data):
+    rec_movies = []
+    user_unwatched = get_friends_unique_watched(user_data)

```

Comment:
[naming] I appreciate these names, since they help me keep straight how the logic works.